### DISCUSSION TITLE
I am thinking of creating a Chrome extension to browse the zen-fs file system in a visual way, just like Finder

### DISCUSSION THREAD

**Discussion by rxliuli** [zen-fs/core#134] [Ideas]
It's just an idea. Currently, there is no intuitive way to browse the file structure. Perhaps an extension that adds an extra panel in the developer tools would be helpful. My question is, if this extension is created, can we modify @zen-fs/core to achieve zero-configuration usage? Because right now, zen-fs does not expose any global variables, like `__zenfs__` or similar.

---

Published https://github.com/rxliuli/zenfs-viewer, Waiting for Chrome Web Store review.
![image](https://github.com/user-attachments/assets/55e5dc8d-e714-4270-94cc-0591b7c97def)

**Comment by james-pre** [comment]
Hi @rxliuli! At the moment, the closest thing we have is the [playground](https://zenfs.dev/playground). I would love to see an extension though! While ZenFS doesn't expose any globals at the moment, I can certainly add one. `__zenfs__` seem appropriate since the global isn't meant to be used.

Edit: See https://github.com/zen-fs/core/commit/8c1ed2114111aa348beedd515cf59535a7689fa4.

**Reply by vednig** [reply]
How does terminal in playground work though, just curious?

**Reply by james-pre** [reply]
@vednig It uses xterm along with some environment related stuff. In the playground [repository](https://github.com/zen-fs/playground), you can find [`/src/shell.ts`](https://github.com/zen-fs/playground/blob/main/src/shell.ts), which has the handling for the shell. [`/system`](https://github.com/zen-fs/playground/tree/main/system) contains files that are available using the `Fetch` backend.

Note that as of right now, there is not support for actually mounting fstabs, though I'd like to support it in the future.

**Reply by rxliuli** [reply]
Published <https://github.com/rxliuli/zenfs-viewer>, Waiting for Chrome Web Store review.
![image](https://github.com/user-attachments/assets/bc04dc2d-f935-463a-abe2-a3ebd7afc93a)

**Reply by james-pre** [reply]
That's awesome!

**Reply by rxliuli** [reply]
Chrome WebStore Reviewed. https://chromewebstore.google.com/detail/zenfs-viewer/mdbgdjlikgdjeandfkabaleddcpaigkk

**Reply by james-pre** [reply]
@rxliuli While this is really cool, my only concern is that uses may think it is official (especially due to the logo usage). I'd really appreciate if you could clarify that it is not official to prevent confusion. Thanks! Also, I can't wait to see this available for Firefox!

**Reply by rxliuli** [reply]
Updated

<img width="770" alt="image" src="https://github.com/user-attachments/assets/b1fd714a-3529-4661-8559-7cf53044b685">
